### PR TITLE
Custom DNS implementation #120

### LINKED ISSUE #120: New-CmAzIaasNetworking doesn't allow you to specify custom dns settings
any network that has a domain controller, you would then need to specify a custom dns server - cloudmarque doesn't allow you to specify one and if you manually configure one whenever you re-run the cmdlet it puts it back to default

### PR DESCRIPTION
**Pull request by deepend-dev**
resolves #120 

- added another parameter dnsServers to support external DNS.
- Empty parameter defaults to Azure DNS.
- Array of IPs can be provided using | for address Space and DNS servers.

### CODE REVIEW

File: Cloudmarque.Azure/Public/IaaS/networking/New-CmAzIaasNetworking.ps1 [hunk 1]
```diff
@@ -236,6 +249,7 @@
 					$vnetObject = @{
 						vnetName     = $vnetName;
 						addressSpace = $addressSpace;
+						dnsServers   = $dnsServers ;
```
Comment (tm-luke) [inline Cloudmarque.Azure/Public/IaaS/networking/New-CmAzIaasNetworking.ps1:252]: remove space here.

File: Cloudmarque.Azure/Public/IaaS/networking/New-CmAzIaasNetworking.json [hunk 2]
```diff
@@ -96,6 +96,7 @@
                                 "addressSpace": {
                                     "addressPrefixes": "[parameters('VnetArmObject')[copyIndex('vnetCopy')].addressSpace]"
                                 },
+                                "dhcpOptions": "[if(not(equals(parameters('VnetArmObject')[copyIndex('vnetCopy')].dnsServers, 'none')), json(concat('{\"dnsServers\": [', replace(replace(string(parameters('VnetArmObject')[copyIndex('vnetCopy')].dnsServers), '[', ''), ']', ''), ']}')), json('null'))]",
```
Comment (tm-luke) [inline Cloudmarque.Azure/Public/IaaS/networking/New-CmAzIaasNetworking.json:99]: Can you re-test with setting the dnsServers set to null? That should functionally be the same as json null.

**Review by tm-luke — CHANGES_REQUESTED** [review]

File: Cloudmarque.Azure/Public/IaaS/networking/New-CmAzIaasNetworking.ps1 (hunk 1 above)
Comment (deepend-dev) [reply Cloudmarque.Azure/Public/IaaS/networking/New-CmAzIaasNetworking.ps1:252]: no space

File: Cloudmarque.Azure/Public/IaaS/networking/New-CmAzIaasNetworking.json (hunk 2 above)
Comment (deepend-dev) [reply Cloudmarque.Azure/Public/IaaS/networking/New-CmAzIaasNetworking.json]: changes completed as discussed

File: Cloudmarque.Azure/Public/IaaS/networking/New-CmAzIaasNetworking.ps1 (hunk 1 above)
Comment (tm-luke) [reply Cloudmarque.Azure/Public/IaaS/networking/New-CmAzIaasNetworking.ps1:252]: nothing between ; and dnsServers?

**Review by tm-luke — APPROVED** [review]

**Pull request merged**